runner/src/mill/runner/MillMain.scala

```diff
@@ -432,6 +433,10 @@ object MillMain {
     for ((k, v) <- desiredProps) System.setProperty(k, v)
   }
 
+  def adjustLogbackOutputFile(): Unit =
+    if (!sys.props.contains("mill.log.file"))
+      sys.props("mill.log.file") = s"${OutFiles.out}/mill.log"
```

Comment:
Could you leave a comment about what this is for?

Reply:
Just added a comment. That was necessary to make `OutputDirectoryTests` pass. The library that coursier uses to extract archives logs things with slf4j. So when unpacking the jansi JAR, some logs where sent. Logback is configured in Mill to handle those. But its configuration file hardcoded `out/` as out directory, so that a file with logs was written there no matter what, and the tests checking that out directory customization works broke.

If we stop using coursier to unpack the jansi JAR, that wouldn't be required for this PR. But it's still nice to fix that (we could add a new non-regression test for it then).